Remotion/Core/UnitTests/Context/SafeContextTests/SafeContextTest.Task.ContinueWith.cs

```diff
@@ -0,0 +1,367 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Remotion.Context;
+using static Remotion.UnitTests.Context.SafeContextTests.SafeContextTaskTest.ContinueWithTestFlags;
+
+namespace Remotion.UnitTests.Context.SafeContextTests
+{
+  public partial class SafeContextTaskTest
```

Comment:
I did not try to backtrack the tests semantics, just a big picture sanity check. do you think we should add an explainatory comment block how the test works?